tiledb/sm/rest/rest_client.cc

```diff
@@ -102,10 +102,11 @@ Status RestClient::get_array_schema_from_rest(
     const URI& uri, ArraySchema** array_schema) {
   // Init curl and form the URL
   Curl curlc;
-  RETURN_NOT_OK(curlc.init(config_, extra_headers_));
+  RETURN_NOT_OK(curlc.init(config_, extra_headers_, res_headers_));
   std::string array_ns, array_uri;
   RETURN_NOT_OK(uri.get_rest_components(&array_ns, &array_uri));
-  std::string url = rest_server_ + "/v1/arrays/" + array_ns + "/" +
+  std::string url = (redirect_uri().empty() ? rest_server_ : redirect_uri()) +
```

Comment:
Let's modify `redirect_uri()` to return `rest_server_` rather than an empty string when we don't have the cached URI. That would simplify this to:
```
std::string url = redirect_uri() + "/v1/arrays/" + array_ns + "/" + curlc.url_escape(array_uri);
```

Additionally, let's rename `redirect_uri()` to `rest_uri()` so that the caller doesn't need to know anything about the direction logic.

Reply:
This wasn't fixed.